Lib/test/test_sys.py

```diff
@@ -518,10 +518,12 @@ def thread_function(num_objects):
         NUM_OBJECTS = 1000
         NUM_THREADS = 10
 
-        # 40 seconds should be enough for the test to be executed: if it
-        # is more than 40 seconds it means that the process is in deadlock
-        # hence the test fails
-        TIMEOUT = 40
+        # 160 seconds should be enough for the test to be executed: if it
+        # is more than 160 seconds it means that the process is in deadlock
+        # hence the test fails.
+        # The timeout is high because on older platforms it takes some time
+        # for the test to be executed when CPython is built with debug flags.
+        TIMEOUT = 160
```

Comment:
1) Consider adding `@support.requires_resource("cpu")` if this test can take a long time to run.
2) You should use `support.SHORT_TIMEOUT` or `support.LONG_TIMEOUT`, which scales with `--timeout`, so it's longer on slow buildbots. 

SHORT_TIMEOUT: It's 30 seconds for `make test` and anywhere from 60 seconds - 135 seconds on the three failing buildbots.

LONG_TIMEOUT: 10x SHORT_TIMEOUT.

If `SHORT_TIMEOUT` isn't enough, you should really consider making the test faster!

Reply:
Thanks for the suggestions. I've actually found a way to make it faster :)

Reply:
Done, let's see how it goes.